tests/helpers/test_frame.py

```diff
@@ -0,0 +1,56 @@
+"""Test the frame helper."""
+import pytest
+
+from homeassistant.helpers import frame
+
+from tests.async_mock import Mock, patch
+
+
+async def test_extract_frame_integration(caplog):
+    """Test extracting the current frame from integration context."""
+    correct_frame = Mock(
+        filename="/home/paulus/homeassistant/components/hue/light.py",
+        lineno="23",
+        line="self.light.is_on",
+    )
+    with patch(
+        "homeassistant.helpers.frame.extract_stack",
+        return_value=[
+            Mock(
+                filename="/home/paulus/homeassistant/core.py",
+                lineno="23",
+                line="do_something()",
+            ),
+            correct_frame,
+            Mock(
+                filename="/home/paulus/aiohue/lights.py",
+                lineno="2",
+                line="something()",
+            ),
+        ],
+    ):
+        found_frame, integration, path = frame.get_integration_frame()
+
+    assert integration == "hue"
+    assert path == "homeassistant/components/"
+    assert found_frame == correct_frame
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5518609)